src/ripple/overlay/impl/Handshake.cpp

```diff
@@ -233,15 +233,15 @@ verifyHandshake(
 {
     if (auto const iter = headers.find("Server-Domain"); iter != headers.end())
     {
-        if (!isProperlyFormedTomlDomain(std::string{iter->value()}))
+        if (!isProperlyFormedTomlDomain(iter->value()))
             throw std::runtime_error("Invalid server domain");
     }
 
     if (auto const iter = headers.find("Network-ID"); iter != headers.end())
     {
         std::uint32_t nid;
 
-        if (!beast::lexicalCastChecked(nid, std::string{iter->value()}))
+        if (!beast::lexicalCastChecked(nid, std::string_view{iter->value()}))
```

Comment:
Are you sure this cast is necessary? Why is it not necessary on line 236?

Reply:
Without the explicit cast, this is the error message from the compiler:
`In template: implicit instantiation of undefined template 'beast::detail::LexicalCast<unsigned int, boost::core::basic_string_view<char>>'`

It looks like boost::string_view is not implicitly convertible to std::string_view during template overload resolution

Reply:
I think it might be implicitly convertible (which is why line 236 works), but this error is a missing specialization of `beast::detail::LexicalCast`. The language won't try other specializations to see if one might work.

Reply:
yeah. i thought of adding one more template specialization for `LexicalCast<Out, boost::string_view>`. but it adds one more abstraction, I can as well convert the type at the point of function call instead of passing it through multiple abstractions

Reply:
Without going into detail, these should be zero-cost (at runtime) abstractions. I prefer to add the missing specialization in one place than a cast at every call site.

Reply:
okay, yeah I didn't think about the runtime costs. fixed in cee5ff89d714229d417e6bbfde6e40864df1d195